[Excel] (custom functions) converting to the unified manifest (#5861)
OfficeDev/office-js-docs-pr: docs/develop/convert-xml-to-json-manifest.md

1. Update the `"$schema"` and `"manifestVersion"` properties at the top of the file to use either the latest released version or the preview version. The following are examples:

   ```json
   "$schema": "https://developer.microsoft.com/json-schemas/teams/v1.30/MicrosoftTeams.schema.json#",

>    "manifestVersion": "1.30",

Reviewer: Nit: Should we use the preview version instead, so that we don't have to keep updating this to use the latest version?

Author: I thought about that, but they're using this to convert a production add-in, which can't have the preview schema, so we'd still have to have a step telling them to update it to the latest released version.

1. Navigate to the [`"developer"`](/microsoft-365/extensibility/schema/root-developer) property and ensure there are child `"privacyUrl"` and `"termsOfUseUrl"` properties. These properties must have appropriate values with a localhost domain.
1. You can now [sideload the add-in](#sideload-the-add-in).

### Special considerations when converting an add-in with custom functions

> Currently neither of the tools described in the [Conversion tools and options](#conversion-tools-and-options) add the needed custom function support to the unified manifest. We're working hard to update the tools. In the meantime, you must add the [`"extensions.runtimes.customFunctions"`](/microsoft-365/extensibility/schema/extension-runtimes-array#customFunctions-property) property manually. Use the [JSON metadata file](../excel/custom-functions-json.md) from your add-in as the source of data to populate the [`"customFunctions"`](/microsoft-365/extensibility/schema/extension-custom-functions) object. The schema for the `"customFunctions"` property is very similar to the schema for JSON metadata file, but not identical. So, while you cannot simply paste the contents of the metadata file into the value of the `"customFunctions"` property, you can cut and paste substantial parts of it. 

Reviewer: ```suggestion
Currently, neither of the tools described in the [Conversion tools and options](#conversion-tools-and-options) section add the needed custom function support to the unified manifest. We're working hard to update the tools. In the meantime, you must add the [`"extensions.runtimes.customFunctions"`](/microsoft-365/extensibility/schema/extension-runtimes-array#customFunctions-property) property manually. Use the [JSON metadata file](../excel/custom-functions-json.md) from your add-in as the source of data to populate the [`"customFunctions"`](/microsoft-365/extensibility/schema/extension-custom-functions) object. The schema for the `"customFunctions"` property is very similar to the schema for JSON metadata file, but not identical. So, while you can't simply paste the contents of the metadata file into the value of the `"customFunctions"` property, you can cut and paste substantial parts of it. 
```

> Some combinations of Office versions and platforms don't yet support the unified manifest. See [Unified manifest - Client and platform support](unified-manifest-overview.md#client-and-platform-support). Currently, to keep your custom functions available on these Office versions, you must [maintain both your existing add-in with the add-in only manifest and your new add-in](../concepts/duplicate-legacy-metaos-add-ins.md). We are working on a system that will enable add-ins that use the unified manifest to be installable on these Office versions. When the system is available, you can unpublish the original add-in. To future proof your new add-in for when that system is available, you should include the optional [metadataUrl](/microsoft-365/extensibility/schema/extension-custom-functions#metadataurl-1) property in the `"customFunctions"` object, and give it the same value that is used in the resource string that is referenced in the [Metadata](/javascript/api/manifest/metadata) element of the add-in only manifest. You should also keep the metadata file in the unified manifest version of the add-in. 

Reviewer: ```suggestion
Some combinations of Office versions and platforms don't yet support the unified manifest. See [Unified manifest - Client and platform support](unified-manifest-overview.md#client-and-platform-support). Currently, to keep your custom functions available on these Office versions, you must [maintain both your existing add-in with the add-in only manifest and your new add-in](../concepts/duplicate-legacy-metaos-add-ins.md). We're working on a system that will enable add-ins that use the unified manifest to be installable on these Office versions. When the system is available, you can unpublish the original add-in. To future proof your new add-in for when that system is available, you should include the optional [metadataUrl](/microsoft-365/extensibility/schema/extension-custom-functions#metadataurl-1) property in the `"customFunctions"` object, and give it the same value that's used in the resource string that's referenced in the [Metadata](/javascript/api/manifest/metadata) element of the add-in only manifest. You should also keep the metadata file in the unified manifest version of the add-in. 
```
